homeassistant/components/vallox/button.py

```diff
@@ -0,0 +1,80 @@
+"""Support for Vallox buttons."""
+from __future__ import annotations
+
+from dataclasses import dataclass
+from datetime import date
+
+from vallox_websocket_api import Vallox
+
+from homeassistant.components.button import ButtonEntity, ButtonEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import ValloxDataUpdateCoordinator, ValloxEntity
+from .const import DOMAIN
+
+
+class ValloxButtonEntity(ValloxEntity, ButtonEntity):
+    """Representation of a Vallox button."""
+
+    entity_description: ValloxButtonEntityDescription
+    _attr_entity_category = EntityCategory.CONFIG
+
+    def __init__(
+        self,
+        name: str,
+        coordinator: ValloxDataUpdateCoordinator,
+        description: ValloxButtonEntityDescription,
+        client: Vallox,
+    ) -> None:
+        """Initialize the Vallox button."""
+        super().__init__(name, coordinator)
+
+        self.entity_description = description
+
+        self._attr_unique_id = f"{self._device_uuid}-{description.key}"
+        self._client = client
+
+
+class ValloxResetFilterChangeDateButton(ValloxButtonEntity):
+    """Representation of a Vallox reset filter button."""
+
+    async def async_press(self) -> None:
+        """Press the button."""
+        await self._client.set_filter_change_date(date.today())
```

Comment:
Since you can manually update the date, wouldn't it make more sense to use a DateEntity? That would make it more useful for example for people who changed the filters last week.

Reply:
Can you also retrieve the last filter change date?

Reply:
Hmm, I received request for a button, but that makes sense. Will change then.

Reply:
True, a DateEntity would be a bit more flexible. 🤔 I liked the simplicity of the button though. OTOH, if one presses it by mistake or forgets to press it after actually switching, it could be useful with a date input.

Reply:
I mean, you can replicate the simplicity of a button by using an automation :)